### ISSUE TITLE
SampleGen: provide empty, instantiated message in request

### ISSUE THREAD

**Issue by beccasaurus** [googleapis/gapic-generator#2837]
The [AutoML API](https://cloud.google.com/natural-language/automl/entity-analysis/docs/reference/rpc/google.cloud.automl.v1beta1) uses an interesting pattern.

They basically have a polymorphic [`Dataset`](https://cloud.google.com/natural-language/automl/entity-analysis/docs/reference/rpc/google.cloud.automl.v1beta1#dataset). The Dataset can be used by...
 - AutoML Vision, where the Dataset can note the [image_classification_dataset_metadata](https://cloud.google.com/vision/automl/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.Dataset.FIELDS.google.cloud.automl.v1beta1.ImageClassificationDatasetMetadata.google.cloud.automl.v1beta1.Dataset.image_classification_dataset_metadata)
 - AutoML Translation, where the Dataset can note the [source and target languages](https://cloud.google.com/translate/automl/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.Dataset.FIELDS.google.cloud.automl.v1beta1.TranslationDatasetMetadata.google.cloud.automl.v1beta1.Dataset.translation_dataset_metadata).
 - ...

[`Dataset`](https://cloud.google.com/translate/automl/docs/reference/rpc/google.cloud.automl.v1beta1#dataset) has separate fields for each AutoML product for them to store their metadata.  
BUT they also use the _existence of the field_ to note which AutoML product the Dataset belongs to.  
AND some of the AutoML's don't actually store any metadata.  
SO they simply created empty messages as placeholders which still serve the purpose of noting which product the Dataset belongs to (and they could always add metadata later).

You can [`filter`](https://cloud.google.com/translate/automl/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.ListDatasetsRequest.FIELDS.string.google.cloud.automl.v1beta1.ListDatasetsRequest.filter) on Datasets by product, e.g. `filter = translation_dataset_metadata:*`. It's a way to get just the Datasets for a particular product.

This •REQUIRES• the ability to [`CreateDataset`](https://cloud.google.com/translate/automl/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.AutoMl.CreateDataset) passing an **empty object** as one of the request parameters.

Here is an [existing code sample](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/automl_language_april_2019/language/automl/automl_natural_language_entity_dataset.py) [published on the website](https://cloud.google.com/natural-language/automl/entity-analysis/docs/datasets#automl-nl-example-python):

```py
def create_dataset(project_id, compute_region, dataset_name):
    """Create a dataset for entity extraction."""
    # [START automl_natural_language_create_dataset]
    # TODO(developer): Uncomment and set the following variables
    # project_id = '[PROJECT_ID]'
    # compute_region = '[COMPUTE_REGION]'
    # dataset_name = '[DATASET_NAME]'

    from datetime import datetime
    from google.cloud import automl_v1beta1 as automl

    client = automl.AutoMlClient()

    # A resource that represents Google Cloud Platform location.
    project_location = client.location_path(project_id, compute_region)

    # Set dataset name and metadata.
    my_dataset = {
        "display_name": dataset_name,
        "text_extraction_dataset_metadata": {}
    }

    # Create a dataset with the dataset metadata in the region.
    dataset = client.create_dataset(project_location, my_dataset)

    # Display the dataset information.
    print("Dataset name: {}".format(dataset.name))
```

---

We need to support this 😕

```yaml
- region_tag: automl_natural_language_entity_create_dataset
  service: AutoMl
  rpc: CreateDataset
  request:
  - field: dataset. text_extraction_dataset_metadata
    value:
```

@googleapis/samplegen

**Comment by beccasaurus** [comment]
ℹHere are the AutoML samples which this is blocking:
 - AutoML Natural Language Entity Extraction.
    - The metadata field on Dataset for this product is [`text_extraction_dataset_metadata`](https://cloud.google.com/natural-language/automl/entity-analysis/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.Dataset.FIELDS.google.cloud.automl.v1beta1.TextExtractionDatasetMetadata.google.cloud.automl.v1beta1.Dataset.text_extraction_dataset_metadata)  
  _(empty message)_
 - AutoML Vision Object Detection
    - The metadata field on Dataset for this product is [`image_object_detection_dataset_metadata`](https://cloud.google.com/vision/automl/object-detection/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.Dataset.FIELDS.google.cloud.automl.v1beta1.ImageObjectDetectionDatasetMetadata.google.cloud.automl.v1beta1.Dataset.image_object_detection_dataset_metadata)  
  _(empty message)_
 - AutoML Video Intelligence
    - The metadata field on Dataset for this product is [`video_classification_dataset_metadata `](https://cloud.google.com/video-intelligence/automl/docs/reference/rpc/google.cloud.automl.v1beta1#google.cloud.automl.v1beta1.Dataset.FIELDS.google.cloud.automl.v1beta1.VideoClassificationDatasetMetadata.google.cloud.automl.v1beta1.Dataset.video_classification_dataset_metadata)  
  _(empty message)_

/fyi @merla18

**Comment by hzyi-google** [comment]
Fixed in #2844
